crates/genesis/src/genesis.rs

```diff
@@ -0,0 +1,1479 @@
+
+    /// Set the mix hash of the header.
+    pub fn with_mix_hash(mut self, mix_hash: B256) -> Self {
+        self.mix_hash = mix_hash;
+        self
+    }
+
+    /// Set the coinbase address.
+    pub fn with_coinbase(mut self, address: Address) -> Self {
+        self.coinbase = address;
+        self
+    }
+
+    /// Set the base fee.
+    pub fn with_base_fee(mut self, base_fee: Option<u64>) -> Self {
+        self.base_fee_per_gas = base_fee;
+        self
+    }
+
+    /// Set the excess blob gas.
+    pub fn with_excess_blob_gas(mut self, excess_blob_gas: Option<u64>) -> Self {
+        self.excess_blob_gas = excess_blob_gas;
+        self
+    }
+
+    /// Set the blob gas used.
+    pub fn with_blob_gas_used(mut self, blob_gas_used: Option<u64>) -> Self {
+        self.blob_gas_used = blob_gas_used;
+        self
+    }
+
+    /// Add accounts to the genesis block. If the address is already present,
+    /// the account is updated.
+    pub fn extend_accounts(
+        mut self,
+        accounts: impl IntoIterator<Item = (Address, GenesisAccount)>,
+    ) -> Self {
+        self.alloc.extend(accounts);
+        self
+    }
+}
+
+/// An account in the state of the genesis block.
+#[derive(Clone, Debug, Default, PartialEq, Eq, Serialize, Deserialize)]
+#[serde(deny_unknown_fields)]
+pub struct GenesisAccount {
+    /// The nonce of the account at genesis.
+    #[serde(skip_serializing_if = "Option::is_none", with = "u64_hex_or_decimal_opt", default)]
+    pub nonce: Option<u64>,
+    /// The balance of the account at genesis.
+    #[serde(deserialize_with = "deserialize_json_u256")]
+    pub balance: U256,
+    /// The account's bytecode at genesis.
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub code: Option<Bytes>,
+    /// The account's storage at genesis.
+    #[serde(
+        default,
+        skip_serializing_if = "Option::is_none",
+        deserialize_with = "deserialize_storage_map"
+    )]
+    pub storage: Option<HashMap<B256, B256>>,
+}
+
+impl GenesisAccount {
+    /// Set the nonce.
+    pub fn with_nonce(mut self, nonce: Option<u64>) -> Self {
+        self.nonce = nonce;
+        self
+    }
+
+    /// Set the balance.
+    pub fn with_balance(mut self, balance: U256) -> Self {
+        self.balance = balance;
+        self
+    }
+
+    /// Set the code.
+    pub fn with_code(mut self, code: Option<Bytes>) -> Self {
+        self.code = code;
+        self
+    }
+
+    /// Set the storage.
+    pub fn with_storage(mut self, storage: Option<HashMap<B256, B256>>) -> Self {
+        self.storage = storage;
+        self
+    }
+}
+
+impl From<GenesisAccount> for Account {
+    fn from(value: GenesisAccount) -> Self {
+        Account {
+            // nonce must exist, so we default to zero when converting a genesis account
+            nonce: value.nonce.unwrap_or_default(),
+            balance: value.balance,
+            bytecode_hash: value.code.map(keccak256),
+        }
+    }
+}
```

Comment:
@mattsse Deleting Account type would mean , we would have to import Account from reth -primitives? That's not an option i think right?

Reply:
we don't need this conversion anymore, instead we can move this to reth-primitives when we replace the genesis type there